onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -238,6 +241,36 @@ class PlannerImpl {
     return false;
   }
 
+  bool FindReusableGraphInput(const onnxruntime::Node& node, int output_arg_num, MLValueIndex* reusable_input) {
+    // skip if this is main graph
+    if (parent_node_) {
+      // only check a single layer. ie. an Identity/Dropout node connecting graph's input and output
+      auto& graph_inputs = graph_viewer_.GetInputs();
```

Comment:
Can you clarify this comment? Not quite sure what you mean by 'only check a single layer'. FindReusableInput looks at the alias and 'may inplace' maps and I'm not quite translating that into limiting the check to a 'single layer'.

Reply:
This logic only checks one node instead of triversal the whole graph.